### PR TITLE
Clean PlayerEx

### PR DESCRIPTION
**Pull request by Brett208**

### CODE REVIEW

File: Source/PlayerEx.cpp [hunk 1]
```diff
@@ -125,11 +125,7 @@ void PlayerEx::SetSatelliteCount(map_id objectType, int count)
 
 int PlayerEx::GetColorNumber()
 {
-	if (!isInited) {
-		return HFLNOTINITED;
-	}
-
-	return playerArray[playerNum].colorNumber;
+	return isInited ? playerArray[playerNum].colorNumber : HFLNOTINITED;
```
Comment (DanRStevens) [inline Source/PlayerEx.cpp:128]: I don't have a problem with this shorter syntax. I just want to point out that if we change to throwing exceptions, we will effectively be undoing this change. The old return statement will be used, and the removed `if` block will be changed to the throw block/function call.

File: Source/PlayerEx.h [hunk 2]
```diff
@@ -28,15 +28,15 @@ class PlayerEx : public _Player
...
-	int GetNumWorkersRequired();		// [assigned workers and scientists]
-	int GetNumScientistsRequired();		// [assigned to buildings, or researching]
+	int GetNumWorkersRequired(); // Assigned workers and scientists
+	int GetNumScientistsRequired(); // Assigned to buildings or researching
 	int GetNumScientistsAsWorkers();
 	int GetNumScientistsAssignedToResearch();
 	int GetTotalFoodProduction();
 	int GetTotalFoodConsumption();
-	int GetFoodLacking();	// [how much food needed to stop people starving ? Note this can be 0 even if there is a food deficit]
+	int GetFoodLacking(); // How much food if required to stop people from starving. This can be 0 even if there is a food deficit
```
Comment (DanRStevens) [inline Source/PlayerEx.h]: Typo: "if"/"is"?

**Review by DanRStevens — APPROVED** [review]

File: Source/PlayerEx.h (hunk 2 above)
Comment (Brett208) [reply Source/PlayerEx.h]: Thank you for catching the typo.

File: Source/PlayerEx.cpp (hunk 1 above)
Comment (Brett208) [reply Source/PlayerEx.cpp:128]: I wanted to use the ternary operator some.

**Pull request merged**